oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/session/OidcProviderSessionRegistry.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2002-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.oidc.authentication.session;
```

Comment:
I feel like we should move this to the top-level package `org.springframework.security.oauth2.client.oidc`. Similar to how `OAuth2AuthorizedClientService` is at the top-level package `org.springframework.security.oauth2.client`.

Reply:
How about moving it to `org.springframework.security.oauth2.client.oidc.session`? This lines up well with `org.springframework.security.core.session`, where `SessionRegistry` and `SessionInformation` reside.